src/lib/State/index.js

```diff
@@ -1,9 +1,69 @@
+var { generateBlocks } = require("./blocks")
+var { Step } = require("./steps")
 var Chapters = require("../Chapters")
 
-function chapters(currentPath, journey) {
+function processBlock({ journey, subject, situation, isActive }, b) {
+  if (b instanceof Step) {
+    b.isActive = isActive
+    journey.push(b)
+  } else if (typeof b == "string") {
+    console.warn(`string step should no longer be used: ${b}`)
+    journey.push({ isActive, path: b })
+  } else {
+    if (!b.steps) {
+      throw Error("" + b + " (" + (b instanceof Array ? "array" : "?") + ")")
+    }
+    let blockSubject = b.subject
+      ? b.subject(subject, situation)
+      : subject || situation
+    const localActive =
+      isActive &&
+      (!b.isActive || (blockSubject && b.isActive(blockSubject, situation)))
+    b.steps.forEach((s) =>
+      processBlock(
+        { journey, subject: blockSubject, situation, isActive: localActive },
+        s
+      )
+    )
+  }
+}
+
+function generateJourney(situation) {
+  const blocks = generateBlocks(situation)
+
+  function processBlocks({ situation }) {
```

Comment:
why are you creating a function here ? instead of just using it in your try catch ?